### PR TITLE
some cleanups

### PR DESCRIPTION
**Pull request by heckflosse**
Some cleanups related to rawimagesource.cc

### CODE REVIEW

**Comment by Desmis** [comment]
@heckflosse 
Ingo

I look the code, compile and test : all seems to work well :)

Thank you

jacques

**Comment by heckflosse** [comment]
@Desmis I tagged this `critical`, because of the performance issues I dectected and fixed in my latest commit.

@Desmis @Floessie @Beep6581 Any objections to merge this asap

File: rtengine/rawimagesource.cc [hunk 1]
```diff
@@ -1333,47 +1330,47 @@ void RawImageSource::preprocess  (const RAWParams &raw, const LensProfParams &le
 
 
     // Always correct camera badpixels from .badpixels file
-    std::vector<badPix> *bp = dfm.getBadPixels( ri->get_maker(), ri->get_model(), idata->getSerialNumber() );
+    std::vector<badPix> *bp = dfm.getBadPixels(ri->get_maker(), ri->get_model(), idata->getSerialNumber());
```
Comment (Floessie) [inline rtengine/rawimagesource.cc:1333]: Still strange that we get a pointer here. Wouldn't an empty vector do? Or an `std::pair<bool, std::vector<>>`, or a `bool get...(std::vector<>&)`, or a `std::optional<>` look-alike...

File: rtengine/rawimagesource.cc [hunk 2]
```diff
@@ -3421,7 +3417,9 @@ void RawImageSource::HLRecovery_blend(float* rin, float* gin, float* bin, int wi
     const float satthresh = 0.5;
 
     float clip[3];
-    FOREACHCOLOR clip[c] = min(maxave, hlmax[c]);
+    for (int c = 0; c < ColorCount; ++c) {
+        clip[c] = min(maxave, hlmax[c]);
```
Comment (Floessie) [inline rtengine/rawimagesource.cc]: Which `min<>()`?

File: rtengine/rawimagesource.cc [hunk 3]
```diff
@@ -3475,23 +3473,25 @@ void RawImageSource::HLRecovery_blend(float* rin, float* gin, float* bin, int wi
             }
         }
 
-        chratio = (sqrt(sum[1] / sum[0]));
+        chratio = (std::sqrt(sum[1] / sum[0]));
```
Comment (Floessie) [inline rtengine/rawimagesource.cc]: Surplus `()`.

File: rtengine/rawimagesource.cc [hunk 4]
```diff
@@ -3895,1993 +3895,491 @@ void RawImageSource::getRowStartEnd (int x, int &start, int &end)
...
+                    } else if (yc[y][x] < 0.24f) {
+                        nh = 139;
+                    } else if (yc[y][x] < 0.29f) {
+                        nh = 140;
+                    } else if (yc[y][x] < 0.32f) {
+                        nh = 141;
+                    } else if (yc[y][x] < 0.34f) {
+                        nh = 142;
+                    } else if (yc[y][x] < 0.37f) {
+
```
Comment (Floessie) [inline rtengine/rawimagesource.cc:4315]: `SQR` --> `std::sqrt()`?

Comment (Floessie) [inline rtengine/rawimagesource.cc]: Typo.

File: rtengine/rawimagesource.cc [hunk 5]
```diff
@@ -6084,53 +4544,35 @@ void RawImageSource::ItcWB(bool extra, double &tempref, double &greenref, double
...
-    Rangeextand.begin =  4;
-    Rangeextand.end =  77;
-    Rangeextand.ng =  73;
-
-    RangeGreen Rangemax;
-    Rangemax.begin =  0;
-    Rangemax.end =  N_g;
-    Rangemax.ng =  N_g;
+    constexpr RangeGreen Rangestandard = {8, 70};
+    constexpr RangeGreen Rangeextand = {4, 77};
```
Comment (Floessie) [inline rtengine/rawimagesource.cc]: `Rangeextended`?

**Review by Floessie — APPROVED** [review]
Just some quick comments.

Builds fine on Debian Testing AMD64 with GCC 9 and Clang 9, and also on Debian Buster i386 with GCC 8.3.

File: rtengine/rawimagesource.cc (hunk 4 above)
Comment (Floessie) [reply rtengine/rawimagesource.cc:4315]: No idea, why GitHub presents the wrong lines here. Just search for "SQR".

Comment (Floessie) [reply rtengine/rawimagesource.cc]: Search for "Weirich".

Comment (heckflosse) [reply rtengine/rawimagesource.cc]: Fixed :+1:

File: rtengine/rawimagesource.cc (hunk 5 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc]: I will change that

File: rtengine/rawimagesource.cc (hunk 4 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc:4315]: SQR squares

File: rtengine/rawimagesource.cc (hunk 2 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc]: fixed

**Comment by Desmis** [comment]
@heckflosse
Ingo
No problem for me :)

jacques

File: rtengine/rawimagesource.cc (hunk 3 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc]: fixed

File: rtengine/rawimagesource.cc (hunk 4 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc:4315]: rt_math.h
```
template<typename T>
constexpr T SQR(T x)
{
    return x * x;
}
```

File: rtengine/rawimagesource.cc (hunk 5 above)
Comment (heckflosse) [reply rtengine/rawimagesource.cc]: fixed

**Pull request merged**